sirepo/pkcli/job_agent.py

```diff
@@ -783,11 +793,20 @@ def start(self):
         self._subprocess.set_exit_callback(self._on_exit)
         return self
 
+    def _assert_run_dir_exists(self):
+        if self.cmd.msg.jobCmd not in ('prepare_simulation', 'compute') \
+           and not self.cmd.run_dir.exists():
```

Comment:
This part could be shared with the other `raise` above. Not critical, but would make the code simpler:

```
if self.cmd.msg.jobCmd not in ('prepare_simulation', 'compute'):
    _assert_run_dir(self.cmd.run_dir)
```